developersample/src/main/java/com/microsoft/identity/client/developersample/MainActivity.java

```diff
@@ -0,0 +1,211 @@
+    @Override
+    public void onSignoutClicked() {
+        mAuthUtil.doSignout();
+        updateSignedOutUI();
+    }
+
+    // Called from GraphDataFragment when the extra scope button is clicked
+    @Override
+    public void onExtraScopeRequested() {
+        mAuthUtil.doExtraScopeRequest(this);
+    }
+
+    // Called from AuthUtil on the failure callback
+    @Override
+    public void onRequestFailure(final MsalException exception) {
+        // Failure UI or mitigation for signin failure should be handled here
+        if (exception instanceof MsalUiRequiredException) {
+            mAuthUtil.doAcquireToken(this, this);
+        }
+    }
+
+    // Called from AuthUtil from the success callback
+    @Override
+    public void useAccessToken(final String accessToken) {
+        Log.d(TAG, "Starting volley request to graph");
+
+        // Make sure we have a token to send to graph
+        if (accessToken == null) {
+            Log.e(TAG, "Null access token was passed");
+            return;
+        }
+
+        new GraphRequestTask(MSGRAPH_URL, accessToken).execute();
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        mAuthUtil = new AuthUtil(this);
+        if (mAuthUtil.getUserCount() == 0) {
+            updateSignedOutUI();
+        } else {
+            mAuthUtil.doAcquireTokenSilent(this);
+        }
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        mAuthUtil.handleInteractiveRequestRedirect(requestCode, resultCode, data);
+    }
+
+    private void updateGraphUI(final String graphResponse) {
+        final Fragment graphDataFragment = GraphDataFragment.newInstance(graphResponse);
+        attachFragment(graphDataFragment);
+    }
+
+    private void attachFragment(final Fragment fragment) {
+        final FragmentManager fragmentManager = getSupportFragmentManager();
+        final FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
+        fragmentTransaction.replace(R.id.activity_main, fragment).addToBackStack(null).commitAllowingStateLoss();
+    }
+
+    private void updateSignedOutUI() {
+        final Fragment signinFragment = SigninFragment.newInstance();
+        attachFragment(signinFragment);
+    }
+
+    private final class GraphRequestTask extends AsyncTask<Void, String, String> {
+
+        private final String mUrl;
+
+        private final String mToken;
+
+        public GraphRequestTask(String url, String token) {
+            mUrl = url;
+            mToken = token;
+        }
+
+        @Override
+        protected String doInBackground(Void... empty) {
+            final URL url;
+            try {
+                url = new URL(mUrl);
+            } catch (final MalformedURLException exc) {
+                Log.e(TAG, "Malformed URL", exc);
+                return null;
```

Comment:
If this returns `null` you can get a `NullPointerException` in `GraphDataFragment#convertJsonToMap(String)`

![screen shot 2017-05-05 at 11 10 55 am](https://cloud.githubusercontent.com/assets/990063/25758645/3a9f7f3a-3184-11e7-92b0-4880cc022ca4.png)